### PR TITLE
Pr darrenfromsandbox

### PR DESCRIPTION
**Pull request by darrenbayliss-dev**
### Change description ###

Build for postgres flexi server in pre

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

**Review by willwatters — CHANGES_REQUESTED** [review]
Please resolve check errors

File: postgres.tf [hunk 1]
```diff
@@ -134,7 +140,7 @@ resource "azurerm_virtual_network_peering" "dg_postgres" {
 # connect data gateway vnet to private dns zone (this will contain the A name for postgres)
 resource "azurerm_private_dns_zone_virtual_network_link" "postgres_dg" {
   name                  = format("%s-%s-virtual-network-link", var.product, var.env)
-  resource_group_name   = var.DNSResGroup
+  resource_group_name   = data.azurerm_resource_group.privatelink_resource_group.name
```
Comment (thomast1906) [inline postgres.tf:143]: whats current issue? this is just swapping string to data return rather than static variable

File: postgres.tf [hunk 2]
```diff
@@ -75,6 +75,12 @@ data "azurerm_subnet" "ss_subnet_pre_postgresql" {
   resource_group_name  = data.azurerm_resource_group.ss_resource_group.name
 }
 
+// Resource Group (this is the resource group where the privatelink.postgres.database.azure.com resides)
+data "azurerm_resource_group" "privatelink_resource_group" {
+  name = "core-infra-intsvc-rg"
```
Comment (thomast1906) [inline postgres.tf:80]: remove unused `var. DNSResGroup` if going with static value

**Comment by thomast1906** [comment]
Just reviewing current error:

```

[2022-11-17T09:19:07.014Z] │ Error: Error: Resource Group "core-infra-intsvc-rg" was not found

[2022-11-17T09:19:07.014Z] │ 

[2022-11-17T09:19:07.014Z] │   with data.azurerm_resource_group.privatelink_resource_group,

[2022-11-17T09:19:07.014Z] │   on postgres.tf line 79, in data "azurerm_resource_group" "privatelink_resource_group":

[2022-11-17T09:19:07.014Z] │   79: data "azurerm_resource_group" "privatelink_resource_group" {
```

You need to add a provider to data resource to reference the subscription.

File: postgres.tf [hunk 3]
```diff
@@ -75,6 +75,12 @@ data "azurerm_subnet" "ss_subnet_pre_postgresql" {
...
+  #id = "/subscriptions/1baf5470-1c3e-40d3-a6f7-74bfbce4b348/resourceGroups/core-infra-intsvc-rg" #invalid key
```
[the remark this answers is not in the document]
Comment (darrenbayliss-dev) [reply postgres.tf:81]: have removed

File: postgres.tf (hunk 1 above)
Comment (darrenbayliss-dev) [reply postgres.tf:143]: yes, I was trying to fix the issue where 2 resource groups have the same name but couldn't fix

File: postgres.tf (hunk 2 above)
Comment (darrenbayliss-dev) [reply postgres.tf:80]: commented this var for now

**Comment by darrenbayliss-dev** [comment]
have added a provider to the resource group, not sure its correct but will test

**Pull request closed without merging**